cmux.go

```diff
@@ -177,7 +177,8 @@ func (l muxListener) Accept() (net.Conn, error) {
 // MuxConn wraps a net.Conn and provides transparent sniffing of connection data.
 type MuxConn struct {
 	net.Conn
-	buf buffer
+	buf     bytes.Buffer
```

Comment:
well, actually I added that line to make a test pass.

I just noted that you have added `err != nil` in writing the HTTP2 test in https://github.com/soheilhy/cmux/blob/bytes-buffer/cmux_test.go#L283 instead of `io.EOF`. Now that it's fixed, there is no need for that micro optimization. WDYT now?